src/components/countries.tsx

```diff
@@ -0,0 +1,92 @@
+import * as React from 'react';
+import { CountriesContext } from '../contexts/countries';
+import { useSorting, useThemeSwitcher } from '../hooks';
+import { CountriesContextType, Language, TypeCountry } from '../@types/country';
+import { ThemeContext } from '../contexts/theme';
+import { ThemeContextType } from '../@types/theme';
+
+type CountriesProps = {
+    selectedContinent: string | null;
+}
+
+const Countries:React.FC<CountriesProps> = ({ selectedContinent }: CountriesProps) => {
+  const { countries } = React.useContext(CountriesContext) as CountriesContextType;
+  const { theme } = React.useContext(ThemeContext) as ThemeContextType;
+  const [ selectedCountry, setSelectedCountry ] = React.useState<string | null>(null);
+  const wrapperRef = React.useRef<HTMLDivElement | null>(null);
+
+  const continentCountries: TypeCountry[] = countries.filter((country: TypeCountry) => country.continent === selectedContinent)
+
+  const firstChars :string[] = Array.from(
+    new Set<string>(
+        continentCountries.map((country: TypeCountry) => {
+          return country.name.charAt(0);
+        })
+    )
+  );
+  useSorting(firstChars);
+  const selectedColor: string = useThemeSwitcher(theme);
+
+  const handleClick = (index: string) => {
+    setSelectedCountry((prev) => {
+      return prev === index ? null : index;
+    }); 
+  };
+
+  React.useEffect(() => {
+    function handleClickOutside(event: MouseEvent) {
+      if (wrapperRef.current && !wrapperRef.current.contains(event.target as Node)) {
+        setSelectedCountry(null);
+      }
+    }
+    document.addEventListener("mousedown", handleClickOutside);
+    return () => {
+      document.removeEventListener("mousedown", handleClickOutside);
+    };
+  }, [wrapperRef]);
+
+  return (
+    <div className='countries' ref={wrapperRef}>
+      {firstChars.length > 0 &&
+        firstChars.map((char: string) => (
```

Comment:
```suggestion
        firstChars.map((char: string) => (
```

I know char is unique, but it's generally a bad key, as it might not be. Keys need to be unique or the world will burn